libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
Instead of completely nuking the error message for something generic, try the Framework's version of the message - https://github.com/joomla-framework/database/blob/master/src/Mysqli/MysqliDriver.php#L516

Also, Framework undeprecated errorMsg and errorNum...